bluemira/equilibria/grad_shafranov.py

```diff
@@ -93,33 +92,36 @@ def __call__(self, nx, nz):
             nz = np.floor_divide(nz + 1, 2)
 
         A = lil_matrix((nx * nz, nx * nz))
-        A.setdiag(ones(nx * nz))
+        A.setdiag(1)
         # NOTE: nb.jit doesn't seem to help here :(
-        for i in range(1, nx - 1):
-            for j in range(1, nz - 1):
-                ind = i * nz + j
-                rp = 0.5 * (x[i + 1] + x[i])  # x_{i+1/2}
-                rm = 0.5 * (x[i] + x[i - 1])  # x_{i-1/2}
-                A[ind, ind] = -(x[i] / d_x2) * (1 / rp + 1 / rm) - 2 / d_z2  # j, l
-                A[ind, ind + nz] = (x[i] / d_x2) / rp  # j, l-1
-                A[ind, ind - nz] = (x[i] / d_x2) / rm  # j, l+1
-                A[ind, ind + 1] = 1 / d_z2  # j-1, l
-                A[ind, ind - 1] = 1 / d_z2  # j+1, l
-            # Apply symmetry boundary if desired
-            if self.force_symmetry:
-                # Apply ghost point method to apply symmetry to (d/dz^2) operator
-                # constributions close to symmetry plane.
-                # If symmetry boundary is centred halfway between cells,
-                # d(psi)/dz = 0 across midplane,
-                # else if symmetry boundary is centred on cells, d(psi)/dz
-                # is equal either side of midplane but reverses sign.
-                ind = i * nz + nz - 1
-                ghost_factor = 1 + nz % 2
-
-                A[ind, ind] = -(x[i] / d_x2) * (1 / rp + 1 / rm) - ghost_factor / d_z2
-                A[ind, ind - 1] = ghost_factor / d_z2
-                A[ind, ind + nz] = (x[i] / d_x2) / rp  # j, l-1
-                A[ind, ind - nz] = (x[i] / d_x2) / rm  # j, l+1
+        i, j = np.meshgrid(np.arange(1, nx - 1), np.arange(1, nz - 1), indexing="ij")
+        i = i.ravel()
+        j = j.ravel()
+        ind = i * nz + j
+        _dx_2_m_halfx = _dx_2 - half_xdx[i]
+        _dx_2_p_halfx = _dx_2 + half_xdx[i]
+        A[ind, ind] = -2 * (_dx_2 + _dz_2)  # j, l
+        A[ind, ind + nz] = _dx_2_m_halfx  # j, l-1
+        A[ind, ind - nz] = _dx_2_p_halfx  # j, l+1
+        A[ind, ind + 1] = _dz_2  # j-1, l
+        A[ind, ind - 1] = _dz_2  # j+1, l
+
+        # Apply symmetry boundary if desired
+        if self.force_symmetry:
+            # Apply ghost point method to apply symmetry to (d/dz^2) operator
+            # contributions close to symmetry plane.
+            # If symmetry boundary is centred halfway between cells,
+            # d(psi)/dz = 0 across midplane,
+            # else if symmetry boundary is centred on cells, d(psi)/dz
+            # is equal either side of midplane but reverses sign.
+            ind = i * nz + nz - 1
+            ghost_factor = 1 + nz % 2
+
+            A[ind, ind] = -(2 + ghost_factor) * _dz_2
+            A[ind, ind - 1] = ghost_factor * _dz_2
+            A[ind, ind + nz] = _dx_2_m_halfx  # j, l-1
+            A[ind, ind - nz] = _dx_2_p_halfx  # j, l+1
```

Comment:
See above - no longer consistent with array used previously.

Reply:
As above - resolved as it's a subtle switch to a different - but still valid - second order discretisation of the problem.